tests/src/com/unciv/uniques/GlobalUniquesTests.kt

```diff
@@ -261,11 +242,92 @@ class GlobalUniquesTests {
+        city.cityConstructions.addBuilding(building.name)
+        city.cityStats.update()
         
         Assert.assertTrue(city.cityStats.finalStatList["Buildings"]!!.science == 30f)
     }
+
+    @Test
+    fun statPercentFromObject() {
+        game.makeHexagonalMap(1)
+        val emptyBuilding = game.createBuildingWithUniques()
+        val civInfo = game.addCiv(
+            uniques = listOf(
+                "[+3 Faith] from every [Farm]",
+                "[+200]% [Faith] from every [${emptyBuilding.name}]",
+                "[+200]% [Faith] from every [Farm]",
+            )
+        )
+        val tile = game.setTileFeatures(Vector2(0f,0f), Constants.desert)
+        val city = game.addCity(civInfo, tile, true)
+        val faithBuilding = game.createBuildingWithUniques()
+        faithBuilding.faith = 3f
+        city.cityConstructions.addBuilding(faithBuilding.name)
+
+        val tile2 = game.setTileFeatures(Vector2(0f,1f), Constants.grassland)
+        tile2.improvement = "Farm"
+        Assert.assertTrue(tile2.getTileStats(city, civInfo).faith == 9f)
+
+        city.cityConstructions.addBuilding(emptyBuilding.name)
+        city.cityStats.update()
+        
+        Assert.assertTrue(city.cityStats.finalStatList["Buildings"]!!.faith == 9f)
+    }
+    
+    @Test
+    fun allStatsPercentFromObject() {
+        game.makeHexagonalMap(1)
+        val emptyBuilding = game.createBuildingWithUniques()
+        val civInfo = game.addCiv(
+            uniques = listOf(
+                "[+3 Faith] from every [Farm]",
+                "[+200]% Yield from every [${emptyBuilding.name}]",
+                "[+200]% Yield from every [Farm]",
+            )
+        )
+        val tile = game.setTileFeatures(Vector2(0f,0f), Constants.desert)
+        val city = game.addCity(civInfo, tile, true)
+        val faithBuilding = game.createBuildingWithUniques()
+        faithBuilding.faith = 3f
+        city.cityConstructions.addBuilding(faithBuilding.name)
+
+        val tile2 = game.setTileFeatures(Vector2(0f,1f), Constants.grassland)
+        tile2.improvement = "Farm"
+        Assert.assertTrue(tile2.getTileStats(city, civInfo).faith == 9f)
+
+        city.cityConstructions.addBuilding(emptyBuilding.name)
+        city.cityStats.update()
+
+        Assert.assertTrue(city.cityStats.finalStatList["Buildings"]!!.faith == 9f)
+    }
     
     
     // endregion
+
+
+    @Test
+    fun statsSpendingGreatPeople() {
+        val civInfo = game.addCiv()
+        val tile = game.setTileFeatures(Vector2(0f,0f), Constants.desert)
+        val cityInfo = game.addCity(civInfo, tile, true)
+        val unit = game.addUnit("Great Engineer", civInfo, tile)
+        val building = game.createBuildingWithUnique("[+250 Gold] whenever a Great Person is expended")
+        cityInfo.cityConstructions.addBuilding(building.name)
+        civInfo.addGold(-civInfo.gold)
+
+        civInfo.addGold(-civInfo.gold) // reset gold just to be sure
```

Comment:
Curious. Nothing can be running in parallel modifying that gold???

Reply:
Nope, just me somehow missing that the above the new line is _exactly the same line_ already. I'll remove it